README.md

```diff
@@ -10,523 +10,568 @@ hard-to-find defects early in the application development lifecycle.
-  the CodeSonar analysis information on the hub.
-* Users can also specify zero or more CodeSonar conditions.
-
-### Required Configuration Settings
-
-| Setting Name     | Description                                                                                                    | Notes                                                                                                                                  |
-|------------------|----------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------|
-| **Protocol**     | The protocol that should be used to communicate with the hub: either http or https.                            | Always explicitly specify https if you are concerned about security, otherwise you may be sending unencrypted data to an imposter hub. |
-|                  | MANUAL: How CodeSonar Works > CodeSonar Structure > Hub > Hub Location                                         |                                                                                                                                        |
-| **Hub address**  | The location of the CodeSonar hub that you are using to manage your analysis results, in format hostname:port. | If you have set up a `${HUB}` parameter for the job, you can use it here.                                                              |
-|                  | MANUAL: How CodeSonar Works > CodeSonar Structure > Hub > Hub Location                                         |                                                                                                                                        |
-| **Project name** | The CodeSonar project name.                                                                                    | If you have set up a `${PROJNAME}` parameter for the job, you can use it here.                                                         |
-|                  | | |
-
-### Optional Configuration Settings
-
-The hub user account you specify here must have sufficient permissions to access the relevant analysis information.
-
-The hub user account credentials that the plugin should use when obtaining analysis information from the hub.
-
-* **MANUAL**: How CodeSonar Works > CodeSonar Structure > Hub > Hub User Accounts
-* **MANUAL**: Role-Based Access Control (RBAC) > RBAC: Role-Permissions
-
-#### Global permissions
-
-You need to configure the CodeSonar user to have the following permissions.
-
-* `G_LIST_PROPERTIES`
-* `G_LIST_USERS`
-* `G_SIGN_IN`
-* `G_SIGN_IN_CERTIFICATE` or `G_SIGN_IN_PASSWORD`, depending on the **Kind of credentials** you are specifying.
+and that it writes Java build output to `/myfiles/buildoutput/classes`, with Java source files located in `/myfiles/src/projX/sources`.
```

Comment:
The generic text that says:

> ...writes Java build output to `/myfiles/buildoutput/classes`, with Java source files located in `/myfiles/src/projX/sources`.

...makes sense in example 4, but not so much in example 5, where the script shows exactly where `/myfiles/buildoutput/classes` is.

> [The example script] writes Java build output to `J.class`, with Java source files located in `.`.

I think that if we show exactly what source files are compiled and how they are compiled (which also indicates where the build output would go), then we should probably use the same source file and build output information in the text description.

I think this leaves two options if we want to address this: use the example as-is, and update the text to accommodate it.  Or change the example to do something non-specific, such as executing a Maven command like `mvn compile`.

The `mvn compile` command finds java source code by a convention where source files are in the subdirectory named 'src' and build output goes in a subdirectory named 'target'.  None of the specific file locations are revealed in the compilation command in this case, so I think with some modifications, our generic text would be applicable and sufficient.

This plugin happens to use Apache Maven, so a Maven command seemed relevant here, but it is just an example and might be too much of a distraction.